CorsixTH/Lua/room.lua

```diff
@@ -429,10 +429,24 @@ function Room:createDealtWithPatientCallback(humanoid)
 end
 
 --! Get the current staff member.
--- Can be overridden in rooms with multiple staff members to return the desired one.
+-- In multi-occupancy rooms this returns the staff member with the minimum service quality
 --!return (staff) The current staff member.
 function Room:getStaffMember()
-  return self.staff_member
+  if not self.staff_member_set then return self.staff_member end
+
+  local staff
+  for staff_member, _ in pairs(self.staff_member_set) do
+    if staff and not staff_member.fired and not staff_member:hasLeavingAction() then
+      if staff:getServiceQuality() > staff_member:getServiceQuality() then
+        staff = staff_member
+      end
+    else
+      if not staff_member.fired and not staff_member:hasLeavingAction() then
```

Comment:
In a room with more staff, if `staff` is not nil but `staff_member:hasLeavingAction()` is true, you first check all conditions at line 439. That fails, and then you check the last 2 again at line 444.

The simpler and cheaper form is to first check the conditions like at line 444, and if these hold, decide whether to also take the service quality into account based on the value of `staff`.